tools/log-collector/log_collector.go

```diff
@@ -79,38 +78,49 @@ func (l *LogCollector) CollectLogsAndDump() error {
 }
 
 // getGVResourcesObjects returns list of objects for given resource_path
-func (l *LogCollector) getResourceObjects(resourcePath string, resource *apiv1.APIResource) (objects unstructured.UnstructuredList) {
+func (l *LogCollector) getResourceObjects(resourcePath string, resource *apiv1.APIResource) (objects unstructured.UnstructuredList,
+	err error) {
 
 	if resource.Namespaced && !l.Clustered {
 		for index := range l.Namespaces {
 			var obj unstructured.UnstructuredList
 			listPath := fmt.Sprintf("%s/namespaces/%s/%s", resourcePath, l.Namespaces[index], resource.Name)
-			err := l.disClient.RESTClient().Get().AbsPath(listPath).Do(context.TODO()).Into(&obj)
+			err = l.disClient.RESTClient().Get().AbsPath(listPath).Do(context.TODO()).Into(&obj)
 			if err != nil {
 				if apierrors.IsNotFound(err) || apierrors.IsForbidden(err) {
 					log.Warnf("%s", err.Error())
-					return objects
+					return objects, nil
 				}
-				return unstructured.UnstructuredList{}
+				if !discovery.IsGroupDiscoveryFailedError(err) {
+					log.Errorf("%s", err.Error())
+					return objects, err
+				}
+				log.Errorf("%s", err.Error())
+				return unstructured.UnstructuredList{}, err
```

Comment:
don't return error for now. We'll update this in future.

Reply:
resolved